### PR TITLE
Add Metrics

### PR DESCRIPTION
**Pull request by harrykeightley**
Add support for metrics.
- Default metrics are "wer" and "cer", but can be extended with parameters in the training job.
- Increase logging around metrics and dataset processing.
- Eval metrics saved under `eval_results.json` post training.

### CODE REVIEW

File: elpis/trainer/trainer.py [hunk 1]
```diff
@@ -45,6 +47,13 @@ def train(
             cache_dir=cache_dir,
             ctc_loss_reduction="mean",
             pad_token_id=processor.tokenizer.pad_token_id,
+            # From https://huggingface.co/blog/fine-tune-xlsr-wav2vec2
```
Comment (benfoley) [inline elpis/trainer/trainer.py]: These commented lines could be removed

Comment (harrykeightley) [reply elpis/trainer/trainer.py]: Removed in latest revision.

File: elpis/datasets/processing.py [hunk 2]
```diff
@@ -60,21 +62,33 @@ def prepare_dataset(dataset: DatasetDict, processor: Wav2Vec2Processor) -> Datas
...
+    columns_to_remove = list(chain.from_iterable(column_names))
+
+    dataset = dataset.map(
+        _prepare_dataset,
+        remove_columns=columns_to_remove,
         num_proc=PROCESSOR_COUNT,
     )
+
+    logger.debug(f"Dataset post prep: {dataset}")
+    logger.debug(f"Training labels: {dataset['train']['labels']}")
```
Comment (benfoley) [inline elpis/datasets/processing.py:93]: Logging the labels here shows an array of vectors. Maybe remove this debug line after sorting out the tokenisation.

**Comment by benfoley** [comment]
LGTM

**Review by benfoley — APPROVED** [review]
LGTM

**Pull request merged**